tests/queries/0_stateless/02003_memory_limit_in_client.expect

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/expect -f
```

Comment:
@FArthur-cmd it seems that this test can be an `sh` test, I converted it here - https://github.com/ClickHouse/ClickHouse/pull/51475

But maybe I missed something? And interactivity is indeed necessary?

Reply:
As far as I remember, this was done to check the expected behavior when running the client. It was necessary to check that the memory limits worked correctly and would not break the client.

Reply:
Ok, then we can convert this to something like:

```
$CLICKHOUSE_CLIENT --max_memory_usage_in_client=1 -n -q "
SELECT arrayMap(x -> range(x), range(number)) FROM numbers(1000); -- { clientError MEMORY_LIMIT_EXCEEDED }
SELECT * FROM (SELECT * FROM system.numbers LIMIT 600000) as num WHERE num.number=60000;
"
```

This should be the same.

But to do this, first need to wait until #51494 will be merged